### PR TITLE
[Website] Restore Android setup instructions in the Getting Started

### PR DESCRIPTION
**Pull request by hramos**
This PR restores some [additional detail](http://facebook.github.io/react-native/releases/0.28/docs/getting-started.html) that was removed from the Android Getting Started this summer.

I'm not fully restoring the original list of steps as the React Native website should not be the source of truth for setting up an Android development environment.

### CODE REVIEW

**Comment by hramos** [comment]
Do not review yet - will be adding Android emulator instructions.

**Comment by facebook-github-bot-9** [comment]
By analyzing the blame information on this pull request, we identified @JoelMarcey and @lacker to be potential reviewers.

**Comment by JoelMarcey** [comment]
@hramos You still want us to hold on the review/accept, right?

**Comment by hramos** [comment]
@JoelMarcey correct. I need to try these instructions out on a clean machine again to be sure we're not missing anything on the Android side.

**Comment by facebook-github-bot-1** [comment]
@hramos updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9867/files/21f0769bd4b26f31eee6143adbc214a1a32b6104..047da894d0dfd7f7319321c5703d0df1e8a9a63a)

**Comment by hramos** [comment]
Android on OS X:

screencapture-localhost-8079-react-native-docs-getting-started-html-1473808679098

Android on Linux:

screencapture-localhost-8079-react-native-docs-getting-started-html-1473808685838

Android on Windows:

screencapture-localhost-8079-react-native-docs-getting-started-html-1473808691878

**Comment by hramos** [comment]
@JoelMarcey I still have to run these steps on a clean OS X install, so I may still end up sending a new PR later, but I think this is good to review for now.

Most of the content being restored here comes from the [0.28 Getting Started guide](http://facebook.github.io/react-native/releases/0.28/docs/getting-started.html#content) you wrote earlier this year, with some edits to redirect users to the Android Studio for more information.

**Comment by bbodenmiller** [comment]
I think the section on making sure the SDK is installed could be simplified to just tell the user to run Android Studio and let Android Studio install the SDK on first run.

I think React Native should be improved to not require `ANDROID_HOME` as Android Studio no longer sets that. Android Studio is still tracking the SDK path somewhere though as it appears in setting dialog boxes even on a fresh project.

Lastly the Windows instructions will still result in the below error and debug failure as React Native expects `adb` to be found in the path and it is not. Something like https://github.com/facebook/react-native/pull/7365 would fix that.

image

**Comment by JoelMarcey** [comment]
@hramos I am good with this partial revert + some modifications. I think it will reduce some confusion that people have had recently. The question is whether we should deal with @bbodenmiller's comments in this pull request, or ship this and then have something separate for those. What do you think?

**Comment by bbodenmiller** [comment]
I vote for shipping this and continuing to address the remaining issues separately.

**Comment by hramos** [comment]
@JoelMarcey I was offline these past few days, apologize for the delay. I'm OK with shipping and addressing followup in a new PR.

**Comment by JoelMarcey** [comment]
@hramos Can you rebase, and then I will ship it? Thanks!

**Comment by facebook-github-bot-5** [comment]
@hramos updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9867/files/047da894d0dfd7f7319321c5703d0df1e8a9a63a..a0aab8b8671e8e76652ec55afd538b3e2d92d2ab)

File: docs/GettingStarted.md [hunk 1]
```diff
@@ -64,115 +64,242 @@ block { display: none; }
...
 
+### Android Development Environment
+
+Setting up your development environment can be somewhat tedious if you're new to Android development. If you're already familiar with Android development, there are a few things you may need to configure. In either case, please make sure to carefully follow the next few steps.
+
+#### 1. Install Android Studio
+
 Download and install [Android Studio](https://developer.android.com/studio/install.html).
 
-If you plan to make changes in Java code, we recommend [Gradle Daemon](https://docs.gradle.org/2.9/userguide/gradle_daemon.html) which speeds up the build.
```
Comment (bbodenmiller) [inline docs/GettingStarted.md]: Looks like this got dropped during the update, was that intentionally?

**Comment by bbodenmiller** [comment]
Should I file my comments above as new issues or how did you guys want to proceed and make sure they don't get lost? Specially while these instructions are an improvement on Windows, it still doesn't work out of the box with these details.

**Comment by hramos** [comment]
As this has already landed, I propose opening a new issue to make sure we keep track of any comments.

File: docs/GettingStarted.md (hunk 1 above)
Comment (hramos) [inline docs/GettingStarted.md]: I don't think this belongs in the Getting Started guide. After all, the goal is to get you up and running with a simple React Native app.

If you do intend to make changes to native Java code, then you'd likely read the Native Android Modules doc, which [now has these instructions](https://github.com/facebook/react-native/pull/9867/files/a0aab8b8671e8e76652ec55afd538b3e2d92d2ab#diff-bdf570846f463516068b23131b72eaafR15).

**Comment by mathewaj** [comment]
Hey @hramos, just going through an install on a clean machine myself, there's a couple of things I had to adjust. Firstly, I had to add the Android environment variable to my ~/.bash_profile not ~/.bashrc for it to work, I know previous instructions on this page had referenced both. Secondly when building, I got `failed to find target with hash string 'android-23'`, so I had to go into build.gradle in my application and update both the compileSdkVersion and the buildToolsVersion to the versions Android Studio had installed, 24 and 24.02 in my case. I suppose these docs could specify specific versions to install alternatively. Would you like me to open separate issue?

**Comment by hramos** [comment]
@mathewaj thanks for taking a look at this on a clean machine! Yes, please open new issues. The 23 vs 24 issue seems problematic, and something we should fix with the CLI itself.

**Pull request closed without merging**